src/main/java/org/openhab/binding/zwave/internal/protocol/serialmessage/AddNodeMessageClass.java

```diff
@@ -83,9 +83,9 @@ public ZWaveSerialPayload doRequestStop(boolean complete) {
         ZWaveSerialPayload payload = new ZWaveTransactionMessageBuilder(SerialMessageClass.AddNodeToNetwork)
                 .withPayload(ADD_NODE_STOP).withRequiresData(false).build();
 
-        if (complete) {
-            payload.setCallbackId(0);
-        }
+//        if (complete) {
```

Comment:
This is definately not good. To complete the transaction, the callback ID must be set to 0.

Reply:
Without a callback the message will **always** time out on Wait_Request based on the serialmessage AddNode definitions. The only thing that I can find that needs the Callback (0) are the "TEST" files that I modified to eliminate that need.  The controller does not need it because this message is sent twice, once with a callback at the end of IncludeProtocolDone and the other at IncludeDone without the callback. I previously proposed setting a timeout on the message, like I did with the "remove node stop message", but you suggested to find the root cause which is resolved with allowing a callback ID.  Here is the debug file with a callback
[AddnodeSuccesswithcallback.txt](https://github.com/openhab/org.openhab.binding.zwave/files/6833878/AddnodeSuccesswithcallback.txt)

Also don't know if you saw this, but someone in the forum posted a partial initialization file that shows the Wait_Request cancellation on this message.  https://community.openhab.org/t/aeotec-zw130-wall-switch-not-saving-configuration/121334/3
I'm not saying this caused any of their problems, just that on an OH3 system totally unrelated to mine this command times out also.
https://community.openhab.org/t/aeotec-zw130-wall-switch-not-saving-configuration/121334/3

If you think there is some useful purpose for a Callback(0) and for the message to timeout, My alternative would be to give it a second, rather than 5.  That way it will not shorten the initialization time by about 4 seconds.  Ironically my variable sleep timer PR with a 18 second cap could handle the 4 second delay as long as another command gets sent in 10 seconds

Reply:
> The only thing that I can find that needs the Callback (0) are the "TEST" files that I modified to eliminate that need. 

Really? Please provide the documentation reference that shows this is no longer needed. My reading of the docs clearly states that this MUST be sent to complete the transaction -:

> After receiving an ADD_NODE_STATUS_DONE status callback, the application MUST terminate the inclusion process by calling the AddNodeToNetwork function one more time. This time, the completedFunc parameter MUST be the NULL pointer.

> Being the exception to the rule, an application calling AddNodeToNetwork(ADD_NODE_STOP) to confirm the reception of a ADD_NODE_STATUS_DONE return code MUST specify the NULL pointer for the completedFunc parameter.

> If you think there is some useful purpose for a Callback(0

My understanding, from reading the docs, is that it is required - and therefore "useful". I don't see that we should change the timeout.

Maybe this has been changed in later versions of the documentation. I've not looked at the most recent version so maybe this is what you are referring to and it has changed this interface?

Reply:
First my response for "useful purpose" was made in haste and I apologize as it could be construed as snarky.  It was late and I had a couple of pints, so should have turned the computer off. Again sorry, very unprofessional.

What I got when I originally read the Silabs docs was that all that is required from the controller is an "Ack" to the ADD_NODE_STOP.  Basically what you are describing more precisely.  With my very limited knowledge I was unable to modify the build message when DONE is "true" to suppress the Request encoded in the serialmessage class to handle the exception;
`.withRequiredRequest(false)` or something like it, 

So I went down the path of what is in this PR and when it worked forgot about the Silabs document.  So more accurately the combination of Callback(0) and Request (True) is the root cause of the timeout. I still think something should be done with this conflict as it is taking 4 seconds out the first 5 seconds the node is awake.

Reply:
I'm not disagreeing that this needs to be resolved, but I don't recall this being a problem in the past so I'm a bit surprised it's a problem now since nothing has changed for a long time. I'm very hesitant to simply remove this call as it states it is required in the Silabs docs, and I have a suspicion that is it likely a security issue if inclusion is not disabled.

Really we shouldn't have this issue included here anyway as this issue is about first time battery initialisation and a PR should not include multiple features as it makes things difficult to resolve. Please split this into a separate issue.

Reply:
So I'm probably not doing this correctly, but I changed this file and the other two related "test" files on the main thread yesterday because I did not want the Silabs concern to be an issue.  I reverted to the timeout option (that you also don't like), but at least there is no Silabs compliancy issue.  It is related to the topic because the message will timeout as Request(true) and Callback(0) can't coexist as message parameters.  At the default 5 seconds, the message eats up about 4 seconds of the initial initialization.   I've attached a notated extract with the baseline binding. [extractcurrentbindingSTOPcommandtimeout.txt](https://github.com/openhab/org.openhab.binding.zwave/files/6840832/extractcurrentbindingSTOPcommandtimeout.txt)

I've provided the attached file in it's full form in the forum earlier, but think there should be at least a partial here in the PR for completeness.

I had thought of three options;
1) Comment out Callback(0) - Original idea, but not Silabs Compliant- Removed
2) Shorten timeout of the message - Current edits
3) Change second "STOP" message to override the serial message Request(true), so only a low level Ack completes the command - Beyond my ability (I've tried but just get errors)

I could revert the RemoveNode file if you want as it is off topic, but is has the same issue (callback(0) & Request(True)).

I guess I need to figure out how to resolve this subthread.

Bob